content/articles/multi-label-classification-with-scikit-multilearn/index.md

```diff
@@ -0,0 +1,654 @@
+
+We use the `train_test_split` to split our dataset into two. 70% of the split dataset is used as the train set, and 30% is used as the test set.
+
+```python
+X_train,X_test,y_train,y_test = train_test_split(Xfeatures,y,test_size=0.3,random_state=42)
+```
+
+### Building model
+
+We will build our model using problem transformation techniques.
+
+Let's start with the first technique.
+
+### Binary relevance technique
+
+This is used to convert our multi-label problem to a multi-class problem.
+
+Let's initialize this technique.
+
+```python
+binary_rel_clf = BinaryRelevance(MultinomialNB())
+```
+
+We also add the `MultinomialNB()` method. `MultinomialNB()` is the Naive Bayes algorithm method used for classification. This is important because by converting convert our multi-label problem to a multi-class problem, we need an algorithm to handle this multi-class problem.
+
+#### Model fitting
+
+We fit the model into the dataset to recognize patterns from the dataset and learn on its own.
+
+```python
+binary_rel_clf.fit(X_train,y_train)
+```
+
+After training, this is the output.
+
+```bash
+BinaryRelevance(classifier=MultinomialNB(alpha=1.0, class_prior=None,
+                                         fit_prior=True),
+                require_dense=[True, True])
+```
+
+As shown, it uses the `MultinomialNB()` to handle the converted multi-class problem.
+
+#### Testing our model
+
+We test our model using the test data. We see if the model can make a prediction using the test data.
+
+```pyhthon
+br_prediction = binary_rel_clf.predict(X_test)
+```
+
+The get the prediction output in the form of an array, use this command.
+
+```python
+br_prediction.toarray()
+```
+
+The output is shown.
+
+```bash
+array([[1., 0., 1.],
+       [1., 0., 1.],
+       [1., 1., 0.],
+       [1., 0., 1.],
+       [1., 0., 1.],
+       [1., 0., 1.],
+       [1., 0., 1.],
+       [1., 1., 0.],
+       [1., 1., 0.],
+       [1., 0., 1.],
+       [1., 0., 1.],
+       [1., 0., 1.],
+       [1., 0., 1.],])
+```
+
+The above output shows how our model has classified our different text into three categories. If a text belongs to that particular category it's assigned `1`; if it does not belong, it's assigned `0`.
+Let's calculate the accuracy score.
+
+#### Accuracy score
+
+```python
+accuracy_score(y_test,br_prediction)
+```
+
+The output:
+
+```bash
+0.9090909090909091
+```
+
+This shows that our model has an accuracy score of `90.91`. This is a good accuracy for our model, and our model has a higher chance of making accurate predictions.
+
+#### Getting Hamming loss
+
+Hamming loss is used to determine the fraction of incorrect predictions of a given model. The lower the hamming loss, the better our model is at making predictions.
+
+```python
+hamming_loss(y_test,br_prediction)
+```
+
+The output is as shown.
+
+```bash
+0.06060606060606061
+```
```

Comment:
What does this result mean? Is the hamming loss low? Explain the result for the benefit of the reader